### PR TITLE
replace NaturalExpDecay, ExponentialDecay, InverseTimeDecay with 2.0 version

### PR DESCRIPTION
**Pull request by longranger2**
### PR types
Others

### PR changes
APIs

### Description
- [x] remove the NaturalExpDecay in python/paddle/fluid/dygraph/learning_rate_scheduler.py, the natural_exp_decay in python/paddle/fluid/layers/learning_rate_scheduler.py  and use NaturalExpDecay in python/paddle/optimizer/lr.py to replace them.
- [x] remove the ExponentialDecay in python/paddle/fluid/dygraph/learning_rate_scheduler.py and use ExponentialDecay in python/paddle/optimizer/lr.py to replace it.
- [x] remove the InverseTimeDecay in python/paddle/fluid/dygraph/learning_rate_scheduler.py and use InverseTimeDecay in python/paddle/optimizer/lr.py to replace it.

### CODE REVIEW

File: python/paddle/distributed/passes/ps_server_pass.py [hunk 1]
```diff
@@ -112,8 +111,8 @@ def _get_lr_scheduler_program(self, lr_scheduler, lr_decay_steps):
             with paddle.static.program_guard(
                 decay_main_program, decay_startup_program
             ):
-                lr = natural_exp_decay(
-                    1.0, lr_decay_steps, lr_scheduler.gamma, True
+                lr = paddle.optimizer.lr.NaturalExpDecay(
+                    1.0, lr_scheduler.gamma
```
Comment (zoooo0820) [inline python/paddle/distributed/passes/ps_server_pass.py:115]: 这个地方的return从一个 Tensor变成Class了，需要检查下能否这么改

File: python/paddle/fluid/layers/learning_rate_scheduler.py [hunk 2]
```diff
@@ -181,68 +180,6 @@ def exponential_decay(learning_rate, decay_steps, decay_rate, staircase=False):
             return decayed_lr
 
 
-def natural_exp_decay(learning_rate, decay_steps, decay_rate, staircase=False):
```
Comment (zoooo0820) [inline python/paddle/fluid/layers/learning_rate_scheduler.py]: 目前看 https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/jit/dy2static/utils.py#L122 函数式LRDecay提供了动转静的功能，这里不建议直接移除。这个PR中可以专注把fluid旧的class类型的LRDecay替换并移除。

**Review by zoooo0820 — APPROVED** [review]
LGTM

**Review by jeff41404 — APPROVED** [review]

**Review by sunzhongkai588 — APPROVED** [review]
no doc'c changes. LGTM

**Pull request merged**